src/main/java/apoc/export/cypher/formatter/CreateCypherFormatter.java

```diff
@@ -13,37 +23,280 @@
+        result.append(CypherFormatterUtils.formatNodeLookup("n1", relationship.getStartNode(), uniqueConstraints, indexedProperties));
+        result.append(", ");
+        result.append(CypherFormatterUtils.formatNodeLookup("n2", relationship.getEndNode(), uniqueConstraints, indexedProperties));
+        result.append(" CREATE (n1)-[r:" + CypherFormatterUtils.quote(relationship.getType().name()));
+        if (relationship.getPropertyKeys().iterator().hasNext()) {
+            result.append(" {");
             result.append(CypherFormatterUtils.formatRelationshipProperties("", relationship, true));
-			result.append("}");
-		}
-		result.append("]->(n2);");
-		return result.toString();
-	}
-}
+            result.append("}");
+        }
+        result.append("]->(n2);");
+        return result.toString();
+    }
+
+    @Override
+    public void statementForNodes(Iterable<Node> nodes, Map<String, Set<String>> uniqueConstraints, Set<String> indexNames, ExportConfig exportConfig, PrintWriter out, Reporter reporter, GraphDatabaseService db) {
+        int batchSize = exportConfig.getBatchSize();
+        ExportConfig.OptimizationType exportType = exportConfig.getOptimizationType();
+
+        AtomicInteger nodeCount = new AtomicInteger(0);
+        Function<Node, Map.Entry<String, Set<String>>> keyMapper = (node) -> {
+            try (Transaction tx = db.beginTx()) {
+                String labels = CypherFormatterUtils.formatAllLabels(node, uniqueConstraints, indexNames);
+                tx.success();
+                Set<String> idProperties = CypherFormatterUtils.getNodeIdProperties(node, uniqueConstraints).keySet();
+                return new AbstractMap.SimpleImmutableEntry<>(labels, idProperties);
+            }
+        };
+        Map<Map.Entry<String, Set<String>>, List<Node>> groupedData = StreamSupport.stream(nodes.spliterator(), true)
+                .collect(Collectors.groupingByConcurrent(keyMapper));
+
+        AtomicInteger propertiesCount = new AtomicInteger(0);
+        AtomicInteger batchCount = new AtomicInteger(0);
+
+        int unwindBatchSize = exportConfig.getUnwindBatchSize();
+        groupedData.forEach((key, nodeList) -> {
+            int nodeListSize = nodeList.size();
+            nodeCount.addAndGet(nodeListSize);
+            for (int i = 0; i < nodeListSize; i++) {
+                writeUnwindStart(exportConfig, out, batchCount);
+                batchCount.incrementAndGet();
+
+                Node node = nodeList.get(i);
+
+                Map<String, Object> props = node.getAllProperties();
+                // start element
+                out.append("{");
+
+                // id
+                Map<String, Object> idMap = CypherFormatterUtils.getNodeIdProperties(node, uniqueConstraints);
+                out.append(idMap.entrySet().stream()
+                        .map(e -> String.format("`%s`: %s", e.getKey(), CypherFormatterUtils.toString(e.getValue()))).collect(Collectors.joining(",")));
```

Comment:
use quote()  and stringbuilder append here instead of string.format

Reply:
ok